### PR TITLE
Update ao and aoco multi segfile tests

### PR DESCRIPTION
**Pull request by kyeap-vmware**
Add more coverage to the ao and aoco test cases by verifying pg_aoseg and pg_aocsseg entries are as expected after upgrade.

pipeline: https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/gpupgrade:additional-ao-testing

### CODE REVIEW

File: test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/sql/ao_table_multi_segfile.sql [hunk 1]
```diff
@@ -17,3 +19,9 @@ CREATE TABLE ao_users (id integer, name text) WITH (appendonly=true) DISTRIBUTED
 
 1:END;
 2:END;
+
+-- Check data
+SELECT * FROM ao_users ORDER BY id;
+
+-- Check table's pg_aoseg entries
+SELECT segno, eof, tupcount, state FROM gp_toolkit.__gp_aoseg_name('ao_users') ORDER BY segno;
```
Comment (jimmyyih) [inline test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/sql/ao_table_multi_segfile.sql:27]: This currently only gets the aoseg information for the coordinator.  We need to run something like this:
```
SELECT * FROM gp_toolkit.__gp_aoseg_name('ao_users') ORDER BY segno;
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg_name('ao_users') AS t FROM gp_dist_random('gp_id')) AS x;
```

Similarly, do the same in the AOCO test and in the target_cluster SQL files too.

**Comment by jimmyyih** [comment]
We need to add some more complex (but common) scenarios as well.  Try adding the following scenarios into the tests.

1. Create an AO table with deleted rows that meets compaction threshold.  After upgrade, verify that vacuum compaction works.
```
source_cluster:
=========================
CREATE TABLE ao_vacuum_compact_after_upgrade (a int, b int) WITH (appendonly=true);

1: BEGIN;
1: INSERT INTO ao_vacuum_compact_after_upgrade SELECT i, i FROM generate_series(1,10)i;
2: INSERT INTO ao_vacuum_compact_after_upgrade SELECT i, i FROM generate_series(11,20)i;
1: COMMIT;

DELETE FROM ao_vacuum_compact_after_upgrade WHERE a > 5;
SELECT * FROM ao_vacuum_compact_after_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg_name('ao_vacuum_compact_after_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg_name('ao_vacuum_compact_after_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;

target_cluster:
=========================
SELECT * FROM ao_vacuum_compact_after_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_vacuum_compact_after_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_vacuum_compact_after_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;

SET gp_select_invisible = on;
SELECT * FROM ao_vacuum_compact_after_upgrade;
VACUUM ao_vacuum_compact_after_upgrade;
SELECT * FROM ao_vacuum_compact_after_upgrade;
SET gp_select_invisible = off;

SELECT * FROM gp_toolkit.__gp_aoseg('ao_vacuum_compact_after_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_vacuum_compact_after_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
```
2. Create an AO table with an AO segment in awaiting drop state.  After upgrade, verify that vacuum completes the AO segment drop.  Also verify that the dropped AO segment is usable again.
```
source_cluster:
=========================
CREATE TABLE ao_with_awaiting_drop_state_before_upgrade (a int, b int) WITH (appendonly=true);
INSERT INTO ao_with_awaiting_drop_state_before_upgrade SELECT i, i FROM generate_series(1,10)i;
DELETE FROM ao_with_awaiting_drop_state_before_upgrade;

1: BEGIN;
1: SELECT * FROM ao_with_awaiting_drop_state_before_upgrade;
2: VACUUM ao_with_awaiting_drop_state_before_upgrade;
1: END;

SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg_name('ao_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;

INSERT INTO ao_with_awaiting_drop_state_before_upgrade SELECT i, i FROM generate_series(1,10)i;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg_name('ao_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;

target_cluster:
=========================
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
VACUUM ao_with_awaiting_drop_state_before_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;

1: BEGIN;
2: BEGIN;
1: INSERT INTO ao_with_awaiting_drop_state_before_upgrade VALUES (88, 88);
2: INSERT INTO ao_with_awaiting_drop_state_before_upgrade VALUES (88, 88);
3: INSERT INTO ao_with_awaiting_drop_state_before_upgrade VALUES (88, 88);
1: COMMIT;
2: COMMIT;

SELECT * FROM ao_with_awaiting_drop_state_before_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
```
3. Create an AO table with empty AO segments.  After upgrade, they should still be there and can be inserted into.
```
source_cluster:
=========================
CREATE TABLE ao_with_empty_aosegs_before_upgrade (a int) WITH (appendonly=true);
1: BEGIN;
2: BEGIN;
1: INSERT INTO ao_with_empty_aosegs_before_upgrade SELECT generate_series(1,10);
2: INSERT INTO ao_with_empty_aosegs_before_upgrade SELECT generate_series(1,10);
3: INSERT INTO ao_with_empty_aosegs_before_upgrade SELECT generate_series(11,20);
1: COMMIT;
2: COMMIT;

DELETE FROM ao_with_empty_aosegs_before_upgrade;
VACUUM ao_with_empty_aosegs_before_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_empty_aosegs_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg_name('ao_with_empty_aosegs_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
SELECT * FROM ao_with_empty_aosegs_before_upgrade;

target_cluster:
==================================================
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_empty_aosegs_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_with_empty_aosegs_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
SELECT * FROM ao_with_empty_aosegs_before_upgrade;

INSERT INTO ao_with_empty_aosegs_before_upgrade SELECT generate_series(1,10);
SELECT * FROM ao_with_empty_aosegs_before_upgrade;
SELECT * FROM gp_toolkit.__gp_aoseg('ao_with_empty_aosegs_before_upgrade');
SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aoseg('ao_with_empty_aosegs_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
```

**Comment by jimmyyih** [comment]
Let's also move the two tests `ao_table_multi_segfile` and `aoco_table_multi_segfile` into the 2nd parallel test group for both source and target upgradeable schedules since we're adding quite a bit of new connections.

File: test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/sql/aoco_table_multi_segfile.sql [hunk 2]
```diff
@@ -1,19 +1,80 @@
 -- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
 -- SPDX-License-Identifier: Apache-2.0
 
--- Test to ensure that AOCO tables with multiple segfiles can be upgraded
+-- Test to ensure that AOCOCO tables with multiple segfiles can be upgraded
```
[the remark this answers is not in the document]
Comment (kyeap-vmware) [reply test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/sql/aoco_table_multi_segfile.sql]: Yup I caught that mistake when reviewing again.

**Comment by jimmyyih** [comment]
Need to rebase to fix the merge conflict on the schedule files.  Also, it seems like one of the tests are failing (locally and in your dev pipeline):
```
        aoco_table_multi_segfile ... FAILED (8.46 sec)  (diff:0.21 sec)
   test clog_preservation        ... ok (38.10 sec)  (diff:0.11 sec)
   test vacuum_freeze_tables     ... ok (1.27 sec)  (diff:0.12 sec)

   =======================
    1 of 27 tests failed.
   =======================

   The differences that caused some tests to fail can be viewed in the
   file "/Users/jyih/workspace/gpupgrade/test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/regression.diffs".  A copy of the test summary that you see
   above is saved in the file "/Users/jyih/workspace/gpupgrade/test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/regression.out".

   running teardown: rm -r /tmp/migration.T5O4t6
   running teardown: archive_state_dir /tmp/gpupgrade.76B5n5

1 test, 1 failure


2023-03-21 17:58:39 ☕  jyihRJGH5 in ~/workspace/gpupgrade
± → cat /Users/jyih/workspace/gpupgrade/test/acceptance/pg_upgrade/5-to-6/upgradeable_tests/source_cluster_regress/regression.diffs
--- \/Users\/jyih\/workspace\/gpupgrade\/test\/acceptance\/pg_upgrade\/5\-to\-6\/upgradeable_tests\/source_cluster_regress\/expected\/aoco_table_multi_segfile\.out	2023-03-21 17:43:13.000000000 -0700
+++ \/Users\/jyih\/workspace\/gpupgrade\/test\/acceptance\/pg_upgrade\/5\-to\-6\/upgradeable_tests\/source_cluster_regress\/results\/aoco_table_multi_segfile\.out	2023-03-21 17:43:13.000000000 -0700
@@ -173,8 +173,8 @@
 --------+-------+------------+----------------+----------+-----+------------------+----------+---------------+-------
  (0,3)  | 1     | 0          | 1              | 10       | 0   | 0                | 2        | 3             | 1
  (0,3)  | 1     | 1          | 129            | 10       | 0   | 0                | 2        | 3             | 1
- (0,4)  | 2     | 0          | 2              | 10       | 0   | 0                | 1        | 3             | 1
- (0,4)  | 2     | 1          | 130            | 10       | 0   | 0                | 1        | 3             | 1
+ (0,5)  | 2     | 0          | 2              | 10       | 0   | 0                | 1        | 3             | 1
+ (0,5)  | 2     | 1          | 130            | 10       | 0   | 0                | 1        | 3             | 1
 (4 rows)
 SELECT gp_segment_id, (t).* FROM (SELECT gp_segment_id, gp_toolkit.__gp_aocsseg_name('aoco_with_awaiting_drop_state_before_upgrade') AS t FROM gp_dist_random('gp_id')) AS x;
  gp_segment_id | gp_tid | segno | column_num | physical_segno | tupcount | eof | eof_uncompressed | modcount | formatversion | state

======================================================================
```

**Comment by jimmyyih** [comment]
Seems like your branch is polluted with contents from a different dev branch after that force push.

**Review by jimmyyih — APPROVED** [review]

**Pull request merged**